sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIO.java

```diff
@@ -528,12 +533,13 @@ public boolean advance() throws IOException {
       try {
         Channel channel = connectionHandler.getChannel();
         // we consume message without autoAck (we want to do the ack ourselves)
-        GetResponse delivery = channel.basicGet(queueName, false);
+        Delivery delivery = consumer.poll(1, TimeUnit.SECONDS);
```

Comment:
Restoring the effective functionality prior to https://github.com/apache/beam/pull/9900 where this will wait a maximum of 1 second.